packages/core/src/indexing/database-index-api.ts

```diff
@@ -55,6 +55,7 @@ type IndexedData<DateType> = {
 export class DatabaseIndexApi<DateType = Date | number> {
   private readonly insertionOrder: InsertionOrder
   private modelsToIndex: Array<StreamID> = []
+  private previouslyIndexedModels = []
```

Comment:
Do we need to save this? Since indexing models also hits the database, should we just query it first?

Reply:
We could certainly query the database instead of maintaining another copy. I'll go ahead and implement that.